rust/arrow/src/compute/kernels/boolean.rs

```diff
@@ -628,6 +642,16 @@ mod tests {
         }
     }
 
+    #[test]
+    fn test_boolean_array_kleene_no_remainder() {
+        let n = 1024;
+        let a = BooleanArray::from(vec![true; n]);
+        let b = BooleanArray::from(vec![None; n]);
+        let result = or_kleene(&a, &b).unwrap();
```

Comment:
What do you  think about a test that covers some more cases on the remainder word? Something like

```
let a = BooleanArray::from(vec![Some(true), Some(false), None, Some(true), Some(false), None, Some(true), Some(false), None, Some(true), Some(false), None])
```

along with some corresponding pattern for `b`?

Reply:
This test aims to reproduce the bug. There are a number of unit tests (`test_bool_array_and_kleene_nulls`, `test_bool_array_or_kleene_nulls`, `test_binary_boolean_kleene_kernel`) that cover different cases of bits in the remainder word. Note that if `batch size < 64` all bits are stored in the remainder chunk.